### ISSUE TITLE
nerd-font icons are too small

### ISSUE THREAD

**Issue by terencode** [bhilburn/powerlevel9k#430]
After installing nerd-fonts and using the new nerdfonts-fontconfig mode, the icons, or at least **some** are too small.
![image](https://cloud.githubusercontent.com/assets/12665563/23801455/816f4440-05b0-11e7-8b72-b672cea2fe2f.png)

As we can see here, the home folder icon I replaced by *'\uf2be'* appear too small as well as the load one that I did not change.

EDIT: After some tests, it seems that the *Font Awesome* glyphicons are the one causing problems... Should this be reported upstream?

**Comment by bhilburn** [comment]
So over in #410, @Cpt0r reporting the thing. Interestingly, no one else seemed to have the problem. 

@kaymmm @lcorsini @akrueger - What nerdfont are you using?

@Cpt0r - What nerdfont are you using where you see the small glyphs?

**Comment by terencode** [comment]
I should add that I installed the AUR package [nerds-fonts-complete](https://aur.archlinux.org/packages/nerd-fonts-complete) and I'm using *DroidSansMonoForPowerline Nerd Font Book*

**Comment by lcorsini** [comment]
I have this problem too, just less, if you use two different fonts they
just don't match sometimes, I use terminus as main font and shire tech mono
as glyph font (non-ascii font in iterm2 on mac, and with fontconfig on
Linux)
Il giorno 10 marzo 2017 @ 17:14:03, Ben Hilburn (notifications@github.com)
ha scritto:

**Comment by akrueger** [comment]
Yes, among all the nerd fonts, the sizes vary and some seem more correct than others.

**Comment by bhilburn** [comment]
I wonder if @ryanoasis has any insight, here.

@ryanoasis - Is there a rule or pattern to how the glyphs are sized in the different nerdfonts?

**Comment by Cpt0r** [comment]
I'm using just one font `Hurmit Medium Nerd Font Complete Mono` on a Mac in iTerm2.

The size issue seems to be with the font itself - different font packs are of different sizes.

Attached is a sample at size 24 relative to normal text.  Some are impossible to see even at that huge size.

<img width="609" alt="screen shot 2017-03-11 at 10 40 23" src="https://cloud.githubusercontent.com/assets/8690075/23817466/7bf8aaaa-0647-11e7-9e43-8cab5327aa45.png">

<img width="607" alt="screen shot 2017-03-11 at 10 44 24" src="https://cloud.githubusercontent.com/assets/8690075/23817512/c0ef5c12-0647-11e7-8d36-7644c70d104c.png">

<img width="615" alt="screen shot 2017-03-11 at 10 39 38" src="https://cloud.githubusercontent.com/assets/8690075/23817473/84e28fbe-0647-11e7-84e1-2a8ce5fa550c.png">

**Comment by ryanoasis** [comment]
@bhilburn Sure no problem.
As for a particular rule nothing as such except that Mono fonts will naturally be smaller glyphs besides that we try to center most of the glyphs and maintain ratios. The patcher has been improved a lot but there are still some issues to work out.

@Cpt0r Those extremely small ones are there even in the original source (from [Devicons](https://github.com/vorillaz/devicons)). As you can see that is how they are in the original source font. Also interesting to note they do not seem to be official glyphs (if you check the [official site](http://vorillaz.github.io/devicons/#/cheat)) but yet they are in there :thinking: 

see below:
![selection_008](https://cloud.githubusercontent.com/assets/8083459/23819794/73474394-05d9-11e7-8595-4b0e4744fa13.png)

The glyphs in Nerd Fonts suffixed with "Mono" are naturally going to be smaller to maintain proper ratio but also still be consider single-width.

---

@terencode 
`f2be` glyph is looking okay for me, see below:
![selection_009](https://cloud.githubusercontent.com/assets/8083459/23819859/921bbef2-05da-11e7-9d49-c9959335a358.png)

**Comment by Cpt0r** [comment]
thanks @ryanoasis, that makes sense, didn't notice that they actually fill the same width as other chars.

So I tried using a larger size of the same font for non-ascii chars but that results in ugliness:

<img width="106" alt="screen shot 2017-03-11 at 14 35 51" src="https://cloud.githubusercontent.com/assets/8690075/23820219/2cbfde2e-0668-11e7-8a48-12641e445c77.png">

I thought I'd try to use the non-mono version for non-ascii chars to hopefully get bigger icons at the same font size but ran into a problem: both font types are registered under the same name, so I cannot specify which to use.

<img width="351" alt="screen shot 2017-03-11 at 14 32 36" src="https://cloud.githubusercontent.com/assets/8690075/23820225/481616fc-0668-11e7-93ee-486b78c7ec73.png">
<img width="406" alt="screen shot 2017-03-11 at 14 32 53" src="https://cloud.githubusercontent.com/assets/8690075/23820226/48197676-0668-11e7-907a-27da61a99554.png">

I think it would be useful to suffix 'Mono' to the mono font embedded name.

**Comment by ryanoasis** [comment]
@Cpt0r One interesting thing from your screenshots it looks like your using version 0.9 of nerd fonts. You might want to try your chances with version 1.0 which has a ton of fixes

**Comment by Cpt0r** [comment]
Hmm I downloaded it very recently, and just now grabbed the zip from here:
https://github.com/ryanoasis/nerd-fonts/releases/tag/v1.0.0

Still says v0.9... can you double check it?
<img width="255" alt="screen shot 2017-03-11 at 15 22 50" src="https://cloud.githubusercontent.com/assets/8690075/23820532/a8f8513c-066e-11e7-9ad8-664108a8bc46.png">

**Comment by ryanoasis** [comment]
Hmm ah okay I'll check it tomorrow thanks. Maybe for some reason the Metadata is outdated

**Comment by ryanoasis** [comment]
@Cpt0r Hey yeah sorry about that those are the latest. What happened was that the internal versions [were updated](https://github.com/ryanoasis/nerd-fonts/commit/17f9ee5db6d81f4fab2987a23d76392814ffb375) _after_ all the fonts were re-patched (e.g. [Hurmit](https://github.com/ryanoasis/nerd-fonts/blob/master/patched-fonts/Hermit/Medium/complete/Hurmit%20Medium%20Nerd%20Font%20Complete.otf)) :blush:

**Comment by bhilburn** [comment]
So, I'm a bit last as to what the status of this thread is, now. As I read it, this is basically a, "We can't do anything about this, and neither can @ryanoasis." Is that accurate?

**Comment by kaymmm** [comment]
I'm pretty sure this particular issue is due to the monospaced font being used. If the standard nerdfonts are used then the icons appear to be the correct size, but the monospaced fonts look small to me using at least a couple of different fonts. FWIW, I typically use fantasque.

**Comment by ryanoasis** [comment]
@bhilburn @kaymmm 

Yeah basically as far as the `...Nerd Font Complete Mono` fonts go those are just going to be smaller (except hopefully for powerline symbols - I'd have to check on that) so that they don't appear squished/stretched and scale appropriately.

The Nerd Fonts without the 'Mono' suffix _should_ not have the 'too small' issue.

---

This image should really help (Droid Sans Mono font patched):


| left | center | right |
|--------------------------------|-----------------|---|
| On the left we have `Mono` version from `v0.8.0` | in the middle `Mono` version from `v1.0.0` | on the right is non-Mono version `v1.0.0`

![selection_010](https://cloud.githubusercontent.com/assets/8083459/23927645/566d496a-08f1-11e7-9c1b-27748db5547c.png)

As you can on the left (version 0.8.0) the `Mono` fonts just got stretched and look terrible so the only way to deal with those is to scale in the single width provided. In the middle is the `Mono` in v1.0.0. To have the best looking would be to go with the non-mono version (as long as your terminal supports non-mono fonts).

**Comment by terencode** [comment]
Seems like it has already been reported upstream: https://github.com/ryanoasis/nerd-fonts/issues/127

**Comment by ryanoasis** [comment]
So overall here is my opinion on this :blush::

1. It is not really an issue (as I stated above)
2. It is definitely not an issue with PL9K

**Comment by bhilburn** [comment]
Thanks so much for your input, @ryanoasis! It is much appreciated.

It looks like this issue is a `NOP` on the P9K side, so I'm going to go ahead and close this one down.

**Comment by akrueger** [comment]
Just out of curiosity, is it common for folks using powerline or nerdfonts to use a non-monospaced font? I can only speak for myself, but I would never use anything but a monospaced font in a terminal. I don't mean to sound like a jerk, I am just surprised and maybe don't understand how either project works fully (can non-monospaced glyphs be combined with monospaced numbers letters, symbols? etc). 

So far I have enjoyed both projects a lot and appreciate all the hard work that goes into them... I'm just curious because I would have assumed (perhaps wrongly) that it would be a dealbreaker for more folks to use a non-monospaced font in the terminal.

**Comment by kaymmm** [comment]
@akrueger while you're not wrong about monospaced fonts in general, the terminology used in this thread is slightly confusing and is about something entirely different. All of the nerd fonts are what you would consider monospaced, meaning that each glyph is the same display width. However, many of the nerd font icons utilize double-width characters, similar to some characters in non-English scripts. Many terminal emulators will allow those double-width glyphs to only occupy a single display character (via settings, e.g., iTerm2 and Terminix both have this). Using the double-width glyphs allows the nerd font icons to appear much more visible for a given font size without actually occupying multiple character spaces. All of the other characters are still the same width so the column-row terminal grid is unadulterated. However, ryanoasis was smart and created an alternate set of so-called `mono` nerd fonts with all single-width glyphs to accommodate terminal emulators that don't properly display double-width characters (e.g., Konsole...ugh! so annoying!), however the icons in these font sets are all smaller than their double-width counterparts, and thus the discussion in this thread.

tl;dr: `mono` refers to single-width icons; they're all fixed-width fonts. also, it's a feature, not a bug.

**Comment by terencode** [comment]
Thanks a lot for the clarification. 
I think  it would be good to link to @ryanoasis and @kaymmm explanations from https://github.com/bhilburn/powerlevel9k/wiki/Troubleshooting

**Comment by akrueger** [comment]
@kaymmm makes sense! Thanks for the clarification and great explanation.

**Comment by Cpt0r** [comment]
thanks @kaymmm!

PS: tl;dr belongs on the first line, not the last :P

**Comment by Cpt0r** [comment]
I can confirm that my small icons issue was due to (mis)using the Mono version of the font - perhaps they should be renamed to avoid confusion for others - eg. Hurmit Single/Double Width - or link to @kaymmm's explanation somewhere public.

**Comment by ryanoasis** [comment]
Sorry to add more noise here but I must also echo that @kaymmm's comment was amazingly well put. 

:+1: x100

I wish I could say it better but I know I can't

**Comment by ddnexus** [comment]
IMO, the term "mono" is quite confusing for a variant, since it doesn't expose what is the real difference. What about "single-width" in place of "mono"?

Also... for the powerline-extended fonts the line height is critical or they won't fill the height of the line, so creating a dent with the background. Different fonts have different line heights, and the script seems to keep into consideration only the width when it scales the glyphs, hence it doesn't work with most of the fonts.

Is there any way to tell the script to scale the glyphs based on the line height value in the font?
